src/Transition/CreateNewIDFUsingRulesV23_2_0.f90

```diff
@@ -567,6 +598,194 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
+                  CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                  nodiff=.false.
+                  OutArgs(1:18)=InArgs(1:18)
+                  OutArgs(19:CurArgs-4) = InArgs(23:CurArgs)
+                  CurArgs = CurArgs - 4
+                  CALL WriteOutIDFLines(DifLfn,'ZoneHVAC:WaterToAirHeatPump',CurArgs,OutArgs,NwFldNames,NwFldUnits)
+                  DO Num3 = 1, NumCoilLatentStuff
+                      IF (('ZONEHVAC:WATERTOAIRHEATPUMP' /= CoilLatentStuff(Num3)%ParentType) .OR. (MakeUPPERCase(InArgs(1)) /= CoilLatentStuff(Num3)%ParentName)) CYCLE
+                      nodiff=.false.
+                      CALL GetNewObjectDefInIDD('Curve:Linear',NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                      IF (TRIM(CoilLatentStuff(Num3)%cHeatPumpTimeConst) == '') THEN
+                          latentTau = 60
+                      ELSE
+                          READ(CoilLatentStuff(Num3)%cHeatPumpTimeConst, *) latentTau
+                      END IF
+                      IF (TRIM(CoilLatentStuff(Num3)%cMaxCyclingRate) == '') THEN
+                          latentNmax = 2.5  ! does this need to be converted to seconds?
+                      ELSE
+                          READ(CoilLatentStuff(Num3)%cMaxCyclingRate, *) latentNmax
+                      END IF
+                      latentA = 4 * latentTau * (latentNmax / 3600.0)
+                      latentCd = latentA * (1 - EXP(-1 / latentA))
+                      OutArgs(1) = CoilLatentStuff(Num3)%NewCurveName
+                      WRITE(OutArgs(2), *) (1 - latentCd)
+                      WRITE(OutArgs(3), *) latentCd
+                      OutArgs(4) = "0.0"
+                      OutArgs(5) = "1.0"
+                      OutArgs(6) = "0.0"
+                      OutArgs(7) = "1.0"
+                      OutArgs(8) = "Dimensionless"
+                      OutArgs(9) = "Dimensionless"
+                      CurArgs = 9
+                      CALL WriteOutIDFLines(DifLfn,'Curve:Linear',CurArgs,OutArgs,NwFldNames,NwFldUnits)
+                      EXIT
+                  END DO
+                  Written = .TRUE.
+
+!              CASE('COIL:COOLING:DX:VARIABLESPEED')  ! PR 10043
+!                  Handled above, integrated with the 90.1 metric
+
+              CASE('COIL:COOLING:WATERTOAIRHEATPUMP:PARAMETERESTIMATION')  ! PR 10043
+                  CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                  DO Num3 = 1, NumCoilLatentStuff
+                      IF (('COIL:COOLING:WATERTOAIRHEATPUMP:PARAMETERESTIMATION' /= CoilLatentStuff(Num3)%CoolingCoilType) .OR. (MakeUPPERCase(InArgs(1)) /= CoilLatentStuff(Num3)%CoolingCoilName)) CYCLE
+                      nodiff=.false.
+                      OutArgs(1:27)=InArgs(1:27)
```

Comment:
Is this safe if the incoming object has <27 fields? I think you need to copy 1:CurArgs here.

Reply:
Hmm...so how do we know whether to append them?  Is this OK?

```fortran
OutArgs(1:CurArgs)=InArgs(1:CurArgs)
IF (CurArgs > 27) THEN
    OutArgs(28) = CoilLatentStuff(Num3)%NewCurveName
    OutArgs(29) = CoilLatentStuff(Num3)%cMaxCyclingRate
    OutArgs(30) = CoilLatentStuff(Num3)%cHeatPumpTimeConst
    OutArgs(31) = CoilLatentStuff(Num3)%cHPDelayTime
    CurArgs = 31
END IF
```

If there is less than 27 input, then I'm not sure what the outcome would be.

Reply:
Good question, I believe the field in between will be written as blank, but we'll have to test one to be sure.